src/rgw/rgw_admin.cc

```diff
@@ -5791,8 +5791,7 @@ int main(int argc, const char **argv)
         }
 
         constexpr bool exclusive = true;
-        int ret = rgw::create_zone(dpp(), null_yield, cfgstore.get(),
-                                   exclusive, zone_params);
+        int ret = cfgstore->create_zone(dpp(), null_yield, exclusive, zone_params, nullptr);
```

Comment:
@dang @cbodley We probably want `rgw::create_zone()` here right? Would it just be a wrapper around `cfgstore->create_zone()`? I left that out for now for the sake of discussion

Reply:
yes, `rgw::create_zone()` has all the generic zone creation logic in it. we shouldn't move generic stuff into RadosConfigStore

Reply:
So in the new `RadosConfigStore->create_zone()` I have, the `get_id()` and `get_name()` checks in the beginning of the function and the `write_default_zone_id()` at the end of the function are generic and should be in `rgw::create_zone()`?

Reply:
yes please

Reply:
One other question I had, is why is create_zone() as it's currently implemented creating zones in ConfigStore as opposed to the backend store? Is there some historical background I'm missing for why things are the way they are now?

Reply:
the `ConfigStore` abstraction was created for the project you're working on - so we can load `RGWZoneParams` and its sal config before we know what kind of backend store and filters we need to create

`sal::ConfigStore` is responsible for the global realm/period/zonegroup/zone objects, and `sal::Driver` is responsible for the users/buckets/objects within a given zone

you can find some additional background in comments on https://github.com/ceph/ceph/pull/47679 and https://github.com/ceph/ceph/pull/48482